File: Java/LeetCode/Sort Array By Parity.java
// https://leetcode.com/problems/sort-array-by-parity/

class Solution {

    public int[] sortArrayByParity(int[] nums) {
        return Arrays.stream(nums)
            .boxed()
            .sorted((a, b) -> {
                int parityComparison = Integer.compare(a % 2, b % 2);
                return (parityComparison == 0) ? Integer.compare(a, b) : parityComparison;
            })
            .mapToInt(i -> i)
            .toArray();
    }

}

File: Kotlin/LeetCode/Sort Array By Parity.kt
// https://leetcode.com/problems/sort-array-by-parity/

class Solution {

    fun sortArrayByParity(nums: IntArray): IntArray {
        return nums.sortedWith(compareBy<Int> { it % 2 != 0 }.then(compareBy { it })).toIntArray()
    }

}
